# Session 14 - Chi-Square Test

In [1]:
import numpy as np
from scipy.stats import chi2_contingency, chisquare

## Task 2 — Expected Counts: Gender vs Music Streaming Preference

$$E_{ij} = \dfrac{\text{row total}_i \times \text{column total}_j}{\text{grand total}}$$

In [2]:
observed_music = {
    'Male':   {'Spotify': 12, 'Gaana': 8, 'YouTube Music': 10},
    'Female': {'Spotify': 18, 'Gaana': 7, 'YouTube Music': 15},
}

row_totals = {g: sum(v.values()) for g, v in observed_music.items()}
col_totals = {genre: sum(observed_music[g][genre] for g in observed_music) for genre in observed_music['Male']}
grand_total = sum(row_totals.values())

print('Row totals:', row_totals)
print('Column totals:', col_totals)
print('Grand total:', grand_total)
print()

expected_music = {}
for gender, row_total in row_totals.items():
    expected_music[gender] = {}
    for genre, col_total in col_totals.items():
        expected = (row_total * col_total) / grand_total
        expected_music[gender][genre] = expected
        print(f'{gender} - {genre}: expected = {expected:.3f}')

Row totals: {'Male': 30, 'Female': 40}
Column totals: {'Spotify': 30, 'Gaana': 15, 'YouTube Music': 25}
Grand total: 70

Male - Spotify: expected = 12.857
Male - Gaana: expected = 6.429
Male - YouTube Music: expected = 10.714
Female - Spotify: expected = 17.143
Female - Gaana: expected = 8.571
Female - YouTube Music: expected = 14.286


## Task 3 — Chi-Square Test of Independence (App Choice vs Payment Method)

Observed counts below come directly from the contingency table built in `session14_contingency_table.xlsx` (Task 1): rows = Zomato / Swiggy / Uber Eats, columns = Paytm / PhonePe / Credit Card.

In [3]:
observed = np.array([
    [5, 1, 4],   # Zomato
    [2, 3, 2],   # Swiggy
    [3, 1, 3],   # Uber Eats
])

chi2_stat, p_value, dof, expected = chi2_contingency(observed)

print('Observed:\n', observed)
print('\nExpected (under independence):\n', np.round(expected, 3))
print('\nChi-square statistic:', chi2_stat)
print('Degrees of freedom:', dof)
print('p-value:', p_value)

Observed:
 [[5 1 4]
 [2 3 2]
 [3 1 3]]

Expected (under independence):
 [[4.167 2.083 3.75 ]
 [2.917 1.458 2.625]
 [2.917 1.458 2.625]]

Chi-square statistic: 3.013333333333333
Degrees of freedom: 4
p-value: 0.5555965834754929


**Conclusion:** p ≈ 0.556, well above α = 0.05 — **fail to reject the null hypothesis**. With only 24 responses, there isn't enough evidence to say app choice and payment method are associated; the pattern in this small sample is consistent with the two being independent.

## Task 4 — Chi-Square Goodness of Fit: Flipkart's 70% Preference Claim

Observed: 18 prefer the new product, 12 do not (n = 30). Claimed proportion: 70% prefer.

In [4]:
observed_counts = [18, 12]
expected_counts = [30 * 0.7, 30 * 0.3]

chi2_stat, p_value = chisquare(observed_counts, f_exp=expected_counts)

print('Observed:', observed_counts)
print('Expected (per 70% claim):', expected_counts)
print('Chi-square statistic:', chi2_stat)
print('p-value:', p_value)

Observed: [18, 12]
Expected (per 70% claim): [21.0, 9.0]
Chi-square statistic: 1.4285714285714286
p-value: 0.23199772362873072


**Conclusion:** p ≈ 0.232, above α = 0.05 — **fail to reject the null hypothesis**. The observed split (18/12) doesn't differ significantly from what the 70% claim would predict (21/9), so this survey doesn't give grounds to dispute Flipkart's stated figure — though with only 30 respondents, the test has limited power to detect a smaller true deviation from 70%.